scripts/bottleneck_detector.py

```
@@ -0,0 +1,583 @@
+        self.detection_file = detection_file or Path(".bottleneck_detection.json")
+        self.max_history = max_history
+        self.workflow_steps: Dict[str, WorkflowStep] = {}
+        self.bottleneck_alerts: deque = deque(maxlen=max_history)
+        self.step_durations: Dict[str, List[float]] = defaultdict(list)
+        self.agent_workloads: Dict[str, int] = defaultdict(int)
+        self.step_queue_sizes: Dict[str, List[int]] = defaultdict(list)
+        self._lock = threading.RLock()
+        self.load_detection_data()
+        
+    def start_workflow_step(self, step_id: str, step_name: str, agent_id: str = "",
+                          dependencies: List[str] = None, metadata: Dict[str, Any] = None) -> WorkflowStep:
```

Ruff (0.14.4)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)